homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -49,7 +51,11 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Frontier Silicon platform."""
+    """Set up the Frontier Silicon platform.
+
+    YAML is deprecated, and imported automatically.
+    SSDP discovery is temporarily retained - to be refactor subsequently.
+    """
     if discovery_info is not None:
```

Comment:
In which case are the entities loaded twice? I fail to see what problem you are mentioning?

In case of an SSDP discovery: they are added here in `async_setup_platform` and not in `setup_entry`.
In case of an import of the old YAML: the entity is migrated here but not added. It is subsequently added from `setup_entry`.
In case of an entity added via the config flow: only added via `setup_entry`.

Reply:
> In which case are the entities loaded twice? I fail to see what problem you are mentioning?
> 
> In case of an SSDP discovery: they are added here in `async_setup_platform` and not in `setup_entry`. In case of an import of the old YAML: the entity is migrated here but not added. It is subsequently added from `setup_entry`. In case of an entity added via the config flow: only added via `setup_entry`.

Aah sorry, I apparently forgot the nomenclature for the old way of loading entities. Still good I mentioned it as you should also move the SSDP discovery from the old discovery way to the config flow, there shouldn't be any in-between steps, its either old platform or new way.

Reply:
Ah - that was a recommendation that I made, to reduce the size of the PR.
I felt it was easier to review without SSDP, and then migrate SSDP as a follow-up PR.

Reply:
Ok. And this was the  future config entry PR mentioned? If that is the case I can be ok with that

Reply:
Yes, I'll be creating a PR for the following branch from the moment this one is merged: https://github.com/wlcrs/home-assistant-core/compare/fsapi-0.2.0-update...wlcrs:home-assistant-core:frontier_silicon_ssdp